src/dswx_sar/mosaic_rtc_burst.py

```diff
@@ -763,25 +764,45 @@ def run(cfg):
         epsg_list = []
 
         for ind, input_dir in enumerate(input_list):
-            # find HDF5 metadata
-            metadata_path = glob.glob(f'{input_dir}/*h5')[0]
-            epsg_list.append(read_metadata_epsg(metadata_path)['epsg'])
 
-        epsg_output = majority_element(epsg_list)
+            first_rtc_path_iter = glob.iglob(f'{input_dir}/*_{first_pol}*.tif')
+            first_rtc_path = next(first_rtc_path_iter)
+            if first_rtc_path:
+                rtc_meta = dswx_sar_util.get_meta_from_tif(first_rtc_path)
+                epsg_list.append(rtc_meta['epsg'])
+            else:
+                err_msg = 'RTC files not found.'
+                raise FileExistsError(err_msg)
 
+        epsg_output = majority_element(epsg_list)
+        logger.info('All RTC bursts and associated masks will be mosaicked ',
+                    f'using the ESPG projection designated by {epsg_output}.')
         # for each directory, find metadata, and RTC files.
         for ind, input_dir in enumerate(input_list):
 
             # find HDF5 metadata
-            metadata_path = glob.glob(f'{input_dir}/*h5')[0]
-            metadata_list.append(metadata_path)
-
             layover_path = glob.glob(f'{input_dir}/*mask.tif')
+            temp_mask_path = f'{scratch_path}/layover_{ind}.tif'
+
             if len(layover_path) > 0:
```

Comment:
```suggestion
            # If both `*_mask.tif` and `*.h5` exists in RTC-S1 burst product directory:
            # The metadata in `*_mask.tif` has priority over HDF5 file.
            if len(layover_path) > 0:
```

Leaving the note regarding the code's behavior